mod/state-transition/pkg/core/state_processor_genesis.go

```diff
@@ -106,6 +108,17 @@ func (sp *StateProcessor[
 		}
 	}
 
+	// BeaconKit enforces a cap on the validator set size. If genesis deposits breach
+	// the case we return an error
+	//#nosec:G701 // can't overflow.
+	if uint32(len(deposits)) > sp.cs.GetValidatorsSetCapSize() {
+		return nil, fmt.Errorf("validator set cap %d, deposits count %d: %w",
+			sp.cs.GetValidatorsSetCapSize(),
+			len(deposits),
+			ErrHitValidatorsSetCap,
+		)
+	}
```

Comment:
here I assume deposits in genesis maps one to one with genesis validators and we do not have two different updates for the same validator.
Is it a fair assumption @chuck-bear, @itsdevbear

Reply:
may be necessary to allow multiple deposits per genesis validator if each deposit is capped to some amount like 32